ogre_mesh_viewer.py

```diff
@@ -609,10 +646,18 @@ def shutdown(self):
     import argparse
 
     parser = argparse.ArgumentParser(description="Ogre Mesh Viewer")
-    parser.add_argument("infile", help="path to a ogre .mesh, ogre .scene or any format supported by assimp")
-    parser.add_argument("-c", "--rescfg", help="path to the resources.cfg")
+    parser.add_argument("infile", help="path to a ogre .mesh, ogre .scene or any format supported by assimp", nargs='?')
+    parser.add_argument("-c", "--rescfg", help="path to the resources.cfg", required=False)
+    parser.set_defaults(infile='')
     args = parser.parse_args()
-    app = MeshViewer(args.infile, args.rescfg)
+
+    if args.infile == "":
```

Comment:
I would actually prefer to only have this part.

It fixes a real UX issue when the user starts meshviewer without any arguments (e.g. by clicking on the icon when having the snap installed)
Previously nothing would happen, leaving the user confused.

The open menu inside the app on the other hand is merely duplicating the file-manager. At least on linux using a much worse UI. So I would rather encourage using the file-manager for navigating to the file instead.

Reply:
I understand, but what is the harm in being able to open a file from within the app if the user wants to change the opened file?

Someday perhaps the FileOpenDialog will be native to ImGUI.

Besides it is usually expected from these types of programs to have a open dialog of some sort.

Another thing: it was a lot of work trying to make it work 🙈

Reply:
my opinion on this was not particularly strong :)